leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/send/SendResource.java

```diff
@@ -65,6 +65,11 @@ public void handlePOST(CoapExchange exchange) {
             return;
         }
 
+        // if clientProfile exists but registration cannot be found because of device ip address change
+        if (clientProfile.getRegistration() == null) {
+            exchange.respond(ResponseCode.NOT_FOUND, "no registration found");
+            return;
+        }
```

Comment:
I'm not sure I get this part of code ?

It seems you try to handle when there is no registration for a given device identity.
It this case, should we rather change `DefaultClientProfileProvider` behavior ? it should maybe return `null` if we didn't find any registration ?

Reply:
yep except that maybe we can get the model only if registration is not null ?

Reply:
OK - corrected in [this](https://github.com/JaroslawLegierski/leshan/commit/9496ef01fbd19c296d67b98f9f6ad6f8f6f4531a) commit